Gym integration and general improvements

tsampazk opened the pull request:
Major refactoring and improvements in docstrings

List of main changes:
1. SupervisorEnv now inherits both from gym.Env and Webots controller.Supervisor. All calls to `self.supervisor...`  changed to `self...`
2. Removed all ABC stuff for abstract methods, now every abstract method just raises a NotImplementedError, similar to gym
3. Docstring additions and improvements
4. Changed imports that used `from .a_deepbots_package` to `from deepbots.a_deepbots_package`
5. Default reset implementation now steps the controller once
6. Added gym in requirements.txt

~I added a TODO comment in all get_timestep methods. This refers to changing timestep to a pythonic class property, we can discuss this further.~

ManosMagnus commented on deepbots/robots/controllers/robot_emitter_receiver_csv.py:
```diff
@@ -1,27 +1,43 @@
+            supervisor node
+        :return: The initialized emitter and receiver references
         """
         emitter = self.robot.getEmitter("emitter")
         receiver = self.robot.getReceiver("receiver")
```
We must use the parameter `emitter_name` and `receiver_name`

ManosMagnus commented on deepbots/robots/controllers/robot_emitter_receiver.py:
```diff
@@ -28,30 +27,32 @@ def __init__(self,
             self.timestep = int(self.robot.getBasicTimeStep())
         else:
-            self.timestep = timestep
+            self.timestep = time_step
```
Same as the previous comment

ManosMagnus commented on deepbots/supervisor/controllers/robot_supervisor.py:
```diff
@@ -30,22 +30,29 @@ def __init__(self, time_step=None):
-            self.timestep = int(self.supervisor.getBasicTimeStep())
+            self.timestep = int(self.getBasicTimeStep())
         else:
             self.timestep = time_step
```
This can be changed to `timestep`

ManosMagnus commented on deepbots/supervisor/controllers/supervisor_emitter_receiver.py:
```diff
@@ -39,34 +73,77 @@ def step(self, action):
     def __init__(self,
                  emitter_name="emitter",
                  receiver_name="receiver",
                  time_step=None):
```
This can be changed to `timestep`

tsampazk commented:
Renamed timestep, added @property getter/setter for timestep and added a deprecation warning in get_timestep.

Also fixed the emitter/receiver name initialization argument issue.
